File: src/main/java/com/alibaba/intl/voddemo/models/VideoMetadata.java
package com.alibaba.intl.voddemo.models;

/**
 * Metadata about a video file.
 *
 * @author Alibaba Cloud
 */
public class VideoMetadata {

    private String title;
    private String fileName;
    private long fileSize; // In bytes
    private String description;

    public VideoMetadata() {
    }

    public VideoMetadata(String title, String fileName, long fileSize, String description) {
        this.title = title;
        this.fileName = fileName;
        this.fileSize = fileSize;
        this.description = description;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getFileName() {
        return fileName;
    }

    public void setFileName(String fileName) {
        this.fileName = fileName;
    }

    public long getFileSize() {
        return fileSize;
    }

    public void setFileSize(long fileSize) {
        this.fileSize = fileSize;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    @Override
    public String toString() {
        return "VideoMetadata{" +
                "title='" + title + '\'' +
                ", fileName='" + fileName + '\'' +
                ", fileSize=" + fileSize +
                ", description='" + description + '\'' +
                '}';
    }
}

File: src/main/resources/static/scripts/model/VideoMetadata.js
/**
 * Metadata about a video file.
 *
 * @author Alibaba Cloud
 */
class VideoMetadata {

    /**
     * @param {string} title
     * @param {string} fileName
     * @param {number} fileSize
     * @param {string} description
     */
    constructor(title, fileName, fileSize, description) {
        /** @type {string} */
        this.title = title;

        /** @type {string} */
        this.fileName = fileName;

        /** @type {number} */
        this.fileSize = fileSize;

        /** @type {string} */
        this.description = description;
    }

}